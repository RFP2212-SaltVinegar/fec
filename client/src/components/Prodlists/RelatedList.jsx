import React,{useState} from 'react';
import '../../css/Prodlists/RelatedList.css';

import Card from './Card.jsx';

// product will be obj from api
// add saved property

//action will be obj with 2 properties
//  imageref and function for action, add and delete
//{product, action}

const RelatedList = () => {

  //const [item, setItem] = useState({product})

  return (
    <div>
      <p>RELATED PRODUCTS</p>
      <div className='related-list'>
        <div className='related-list-container'>
          <button >{'<'}</button>
          <div className='related-list-card-container'>
            <Card />
            <Card />
            <Card />
            <Card />
          </div>
          <button>{'>'}</button>
        </div>
      </div>
    </div>
  )
}

export default RelatedList;