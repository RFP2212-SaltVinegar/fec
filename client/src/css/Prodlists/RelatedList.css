.related-list{
  border: 1px solid red;
  text-align: center;

}

.related-list-container{
  border: 1px solid rgb(111, 55, 55);
  display: inline-flex;
  flex-direction: row;
  min-width: 1000px;
  height: 361px;
  text-align: center;
}

.related-list-container button{
  margin-left: 10px;
  margin-right: 10px;
}

.related-list-card-container{
  display: inline-flex;
  flex-direction: row
}

.related-list-container-lscroll{

}

.related-list-container-rscroll{

}