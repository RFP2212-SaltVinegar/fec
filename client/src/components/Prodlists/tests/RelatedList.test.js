import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import axios from 'axios';
import RelatedList from '../RelatedList.jsx';

jest.mock('axios');


describe('RelatedList component', () => {
  let product;

  beforeEach(() => {
    const localStorageMock = {
      getItem: jest.fn().mockImplementation((key) => {
        if (key === 'user') {
          return '{"outfits":[8,4]}';
        }
        return null;
      })
    };
    Object.defineProperty(window, 'localStorage', {
      value: localStorageMock,
    });


    // const getItem = jest.fn((key) => key in localStorage ? localStorage[key] : null)
    // const setItem = jest.fn((key, value) => (localStorage[key] = value))

    product = { id: 1};
    axios.get.mockImplementation(
      (url) => {
        console.log(url);

        const regex1 = /^api\/products\/.*\/related$/g;
        const regex2 = /^api\/products\/.$/g;
        const regex3 = /^api\/products\/.*\/styles$/g;
        const regex4 = /^api\/reviews\/meta$/g;
        // console.log(regex1.test(url))
        switch (true) {
          case regex1.test(url):{
            //console.log('hit')
            return Promise.resolve(
              { data: [2,3,4,5,6,7,8] }
            )
            break;
          }
          case regex2.test(url):{
            return Promise.resolve(
              { data: {
                id: 1,
                name: 'name',
                description: 'description',
                category:'category',
                default_price:'0',
                features: [{feature:'feature',value:'value'}]
              } }
            )
            break;
          }
          case regex3.test(url):{
            return Promise.resolve(
              { data: {
                product_id: '1',
                results: [
                  {
                    'style_id': 1,
                    'photos': [
                      {
                        thumbnail_url:'thumbnail_url',
                        url: 'url'
                      }
                    ]
                  }
                ]
              } }
            )
            break;
          }
          case regex4.test(url):{
            return Promise.resolve(
              { data: {
                product_id: 1,
                ratings:{
                  1: 1,
                  2: 1,
                  3: 1,
                  4: 1,
                  5: 1
                }
              } }
            )
            break;
          }
        }

      }
    );
  });

  it('should render related products', async () => {
    const {findByTestId,findByText,getByText} = render(<RelatedList product={product} />);
    const title = await findByText('Related Products');
    expect(title).toBeInTheDocument();

    const view = await findByTestId('2');
    expect(view).toBeInTheDocument();


    const rightArrow = await findByTestId('arrow-right');
    fireEvent.click(rightArrow);
    const nextView = await findByTestId('7')
    expect(nextView).toBeInTheDocument();
  });
});


