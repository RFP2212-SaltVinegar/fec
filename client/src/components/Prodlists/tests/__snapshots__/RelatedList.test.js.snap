// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div>
  <p
    className="related-list-title"
  >
    Related Products
  </p>
  <div
    className="related-list"
  >
    <div
      className="related-list-container"
    >
      <button
        className="related-list-container-scroll"
        disabled={true}
        onClick={[Function]}
      >
        <i
          className="related-list-arrow-left"
        />
      </button>
      <div
        className="related-list-card-container"
      />
      <button
        className="related-list-container-scroll"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="related-list-arrow-right"
        />
      </button>
    </div>
  </div>
</div>
`;
